src/components/LayerCollection/LayerPOLine.js

```diff
@@ -156,18 +166,40 @@ class LayerPOLine extends Component {
     }
   };
 
-  updatePOLine = (data) => {
+  openOrder = async (saveAndOpen) => {
+    const { mutator } = this.props;
+    const order = this.getOrder();
+    const updatedOrder = await mutator.order.PUT({
+      ...order,
+      workflowStatus: WORKFLOW_STATUS.open,
+    });
+
+    return saveAndOpen
+      ? updatedOrder
+      : Promise.resolve();
+  }
+
+  updatePOLine = ({ saveAndOpen, ...data }) => {
     const line = cloneDeep(data);
-    const { parentMutator, location: { pathname } } = this.props;
+    const { location: { pathname }, parentMutator } = this.props;
 
     parentMutator.poLine.PUT(line)
+      .then(() => this.callout.current.sendCallout({
+        type: 'success',
+        message: <FormattedMessage id="ui-orders.success" />,
+      }))
+      .then(() => this.openOrder(saveAndOpen))
       .then(() => {
         parentMutator.query.update({
           _path: `${pathname}`,
           layer: null,
         });
       })
       .catch(e => this.handleErrorResponse(e, line));
+
+    if (saveAndOpen) {
+      this.openOrder();
```

Comment:
why do you need to invoke it here when it was invoked in promise chain?

Reply:
Just forgot to remove it